RecoVertex/KalmanVertexFit/src/SingleTrackVertexConstraint.cc

```diff
@@ -37,7 +37,8 @@ SingleTrackVertexConstraint::BTFtuple SingleTrackVertexConstraint::constrain(
   typedef VertexTrack<5>::RefCountedLinearizedTrackState RefCountedLinearizedTrackState;
 
   double field  = track.field()->inInverseGeV(track.impactPointState().globalPosition()).z();
-  if (fabs(field) < 1e-4) {
+  double field0  = track.field()->inTesla(GlobalPoint(0.,0.,0.)).z();
```

Comment:
here
field().nominalValue() should be used
(mind is integer and in KG)
http://cmslxr.fnal.gov/lxr/source/MagneticField/Engine/interface/MagneticField.h?v=CMSSW_7_5_0_pre5#0056